docs/pbench-agent.rst

```diff
@@ -0,0 +1,1348 @@
+   yum clean expire-cache
+   yum install pbench-agent
+
+If you upgrade to a release later than -102, please also ensure that you
+clear out your tools and reregister them after the upgrade: the label
+handling has changed. For example:
+
+::
+
+   clear-tools
+   register-tool-set --label=ceph-server --remote=my-ceph-server
+   register-tool-set --label=kvmhost --remote=my-kvmhost
+   register-tool-set --label=kvmguest --remote=my-kvmguest
+
+First steps
+===========
+
+All of the commands take a ``--help`` option and produce a terse usage
+message.
+
+The default set of tools for data collection can be enabled with
+
+::
+
+   register-tool-set
+
+You can then run a built-in benchmark by invoking its pbench script -
+pbench will install the benchmark if necessary [2]_:
+
+::
+
+   pbench_fio
+
+When the benchmark finishes, the tools will be stopped as well. The
+results can be collected and shipped to the standard storage
+location [3]_ with
+
+::
+
+   move-results
+
+or
+
+::
+
+   copy-results
+
+First steps with user-benchmark
+-------------------------------
+
+If you want to run something that is not already packaged up as a
+benchmark script, you may be able to use the ``user-benchmark`` script:
+it takes a command as argument, starts the collection tools, invokes the
+command, stops the collection tools and postprocesses the results. So
+the workflow becomes:
+
+::
+
+   register-tool-set
+   user-benchmark --config=foo -- myscript.sh
+   move-results
+
+See for host in $hosts ;do register-tool-set --remote=$host done user-benchmark --config=foo -- myscript.sh move-results 
+
+Apart from having to register the collection tools on **all** the hosts,
+the rest is the same: ``user-benchmark`` will start the collection tools
+on all the hosts, run ``myscript.sh``, stop the tools and run the
+postprocessing phase, gathering up all the remote results to the local
+host (the local host may be just a controller, not running any
+collection tools itself, or it may be part of the set of hosts where the
+benchmark is run, with collection tools running).
+
+The underlying assumption is that ``myscript.sh`` will run your
+benchmark on all the relevant hosts and will copy all the results into
+the standard directory which postprocessing will copy over to the
+controller host. ``user-benchmark`` calls the script in its command-line
+arguments (everything after the – is just execed by user-benchmark) and
+redirects its ``stdout`` to a file in that directory:
+``$benchmark_run_dir/result.txt``.
+
+Defaults
+========
+
+The benchmark scripts source the base script
+(``/opt/pbench-agent/base``) which sets a bunch of defaults:
+
+::
+
+   pbench_run=/var/lib/pbench
+   pbench_log=/var/lib/pbench/pbench.log
+   date=`date --utc "+%Y.%m.%dT%H.%M.%S"`
+   hostname=`hostname -s`
+   results_repo=pbench@pbench.example.com
+   results_repo_dir=/pbench/public_html/incoming
+   ssh_opts='-o StrictHostKeyChecking=no'
+
+These are now specified in the config file
```

Comment:
This whole section does not sound right: `base` does set a bunch of variables but most of these are not specified in the config file. In particular, `date`, `hostname`, `results_repo` and `results_repo_dir` are not specified in the config file. The last two I can't find anywhere, either in the config file or in `base`.